spring-integration-core/src/main/java/org/springframework/integration/endpoint/EventDrivenConsumer.java

```diff
@@ -46,6 +49,25 @@ public EventDrivenConsumer(SubscribableChannel inputChannel, MessageHandler hand
 		this.setPhase(Integer.MIN_VALUE);
 	}
 
+	@Override
+	public String getInputChannelName() {
+		return this.inputChannel == null ? null : this.inputChannel instanceof NamedComponent ?
+			((NamedComponent) this.inputChannel).getComponentName() : null;
+	}
+
+
+	@Override
+	public String getOutputChannelName() {
+		MessageChannel channel = this.handler instanceof AbstractMessageProducingHandler ?
```

Comment:
We should reconsider such a logic here everywhere, when we don't use interfaces.
In case of proxy, e.g. `@IdempotentReceiver` or `@Transactional` we aren't able to use `instanceof` so easy.

Maybe we just simple move `getOutputChannel()` to the `MessageProducer`?..